app/mailboxes/mailbox_helper.rb

```diff
@@ -35,7 +35,7 @@ def add_attachments_to_message
 
   def create_contact
     @contact_inbox = ::ContactBuilder.new(
-      source_id: "email:#{processed_mail.message_id}",
+      source_id: processed_mail.original_sender,
```

Comment:
Existing conversations would break if we push this directly.

Reply:
@tejaswinichile Can we change the `original_sender` method, how do we consider `message_id` further?

Reply:
I am not sure this has an impact on existing conversations. In the case of email Inboxes, we are fetching contact inbox using a combination of contact & inbox, not source_id.  So this change should help us to make the behavior consistent: "source_id will store the identifier of the contact in the external system"

Reply:
As per the discussion with @aswindevps email's `message_id`  can not be the source_id for the `contact_inbox` we are saving that as a message's source_id which would be the source of the truth for the message.

Updating the source_id for the contact_inbox to the original_sender makes sense here and as we dug through the code it won't have any further impact.